### PR TITLE
Self-updating HookSet transaction added.

### PR DESCRIPTION
**Pull request by Udith-Gayan**

### CODE REVIEW

File: lib/macro.h [hunk 1]
```diff
@@ -553,8 +553,9 @@ int out_len = 0;\
         _06_08_ENCODE_DROPS_FEE            (fee_ptr, fee                            );                               \
     }
 
-
-
 #endif
 
+#define GET_HOOKSET_OPERATION_SIZE(operation_type)\
+    (operation_type == 1U ? 75 : operation_type == 2 ? 9 : 2)
```
Comment (kithminisg) [inline lib/macro.h:559]: It is better if you can introduce these figures using constants. (with proper names)

File: lib/macro.h [hunk 2]
```diff
@@ -553,8 +553,14 @@ int out_len = 0;\
         _06_08_ENCODE_DROPS_FEE            (fee_ptr, fee                            );                               \
     }
 
-
-
 #endif
 
+#define OP_HOOK_INSTALLATION 1
```
Comment (chalith) [inline lib/macro.h:558]: Let's move these to evernode.h because this file is directly taken from xrpld-hooks repo

File: src/constants.h [hunk 3]
```diff
@@ -13,6 +13,7 @@
 #define HOST_REWARD "evnHostReward"
 #define HOST_TRANSFER "evnTransfer"
 #define HOST_REBATE "evnHostRebate"
+#define NEW_SET_HOOK_HASHES "evnNewSetHookHashes"
```
Comment (chalith) [inline src/constants.h:16]: evnHookUpdate?

**Review by kithminisg — APPROVED** [review]

File: src/evernodeone.c [hunk 4]
```diff
@@ -48,264 +48,321 @@ int64_t hook(uint32_t reserved)
...
         GET_MEMO(0, memos, memos_len, memo_ptr, memo_len, type_ptr, type_len, format_ptr, format_len, data_ptr, data_len);
 
-        if (op_type == OP_HOST_REG)
+        if (op_type == OP_HOST_REG || op_type == OP_SET_HOOK)
         {
-            int is_valid = 0;
-            BUFFER_EQUAL_STR_GUARD(is_valid, format_ptr, format_len, FORMAT_TEXT, 1);
-            if (!is_valid)
-                rollback(SBUF("Evernode: Memo format should be text."), 1);
+            if (op_type == OP_HOST_REG)
```
Comment (chalith) [inline src/evernodeone.c:53]: Do we need two if conditions? Wouldn't if else if enough wince there are no common code
```
if (op_type == OP_HOST_REG)
else if (op_type == OP_HOST_REG)
```

**Review by chalith — APPROVED** [review]

**Pull request merged**